### PR TITLE
✨   KS release test OCP: deployment in multiple clusters

### PR DESCRIPTION
**Pull request by dumb0002**
## Summary

This PR adds an easy release testing for KS on OCP for deployment in multiple clusters

## Related issue(s)
https://github.com/kubestellar/kubestellar/issues/2037

Fixes #

### CODE REVIEW

File: test/e2e/common/cleanup.sh [hunk 1]
```diff
@@ -16,12 +16,53 @@
 set -x # echo so that users can understand what is happening
 set -e # exit on error
 
+
+env="kind"
+
+if [ "$1" == "--env" ]; then
+    env="$2"
+    shift
```
Comment (MikeSpreitzer) [inline test/e2e/common/cleanup.sh:24]: Consuming two command line arguments would normally be done with _two_ `shift` commands.
This is not a fatal problem at the moment, because nothing later looks at the command line. But this is a fragile comfort.

File: test/e2e/common/cleanup.sh [hunk 2]
```diff
@@ -16,12 +16,53 @@
 set -x # echo so that users can understand what is happening
 set -e # exit on error
 
+
+env="kind"
+
+if [ "$1" == "--env" ]; then
```
Comment (MikeSpreitzer) [inline test/e2e/common/cleanup.sh:22]: What should happen if there is no `--env` on the command line? I think that a silent "nothing" is not the best answer.

File: test/e2e/multi-cluster-deployment/run-test.sh [hunk 3]
```diff
@@ -18,25 +18,43 @@
...
     setup_flags="$1"
     shift
 fi
 
-if [ "$#" != 0 ]; then
-    echo "Usage: $0 [--released]" >& 2
-    exit 1
+if [ "$1" == "--env" ]; then
+    env="$2"
+    shift
```
Comment (MikeSpreitzer) [inline test/e2e/multi-cluster-deployment/run-test.sh:30]: Consuming `--env $something` requires _two_ `shift` commands.

File: test/e2e/multi-cluster-deployment/run-test.sh [hunk 4]
```diff
@@ -18,25 +18,43 @@
...
-"${HACK_DIR}/check_pre_req.sh" --assert --verbose kubectl docker kind make go ko yq helm kflex ocm
+    "${COMMON_SRCS}/cleanup.sh"
+    source "${COMMON_SRCS}/setup-shell.sh"
+    "${COMMON_SRCS}/setup-kubestellar.sh" $setup_flags
+    "${SRC_DIR}/use-kubestellar.sh"
+fi
+
+
+if [ $env == "ocp" ];then
+   bash <(curl -s https://raw.githubusercontent.com/kubestellar/kubestellar/release-$KUBESTELLAR_VERSION/test/e2e/common/cleanup.sh) --env ocp
```
Comment (MikeSpreitzer) [inline test/e2e/multi-cluster-deployment/run-test.sh:55]: Why the change to fetching the cleanup script from github? In the `kind` case, the local copy is used. These E2E tests are specifically designed to be able to test the local copy of the repo.

File: test/e2e/multi-cluster-deployment/run-test.sh [hunk 5]
```diff
@@ -18,25 +18,43 @@
...
+   source <(curl -s https://raw.githubusercontent.com/kubestellar/kubestellar/release-$KUBESTELLAR_VERSION/test/e2e/common/setup-shell.sh)
+   bash <(curl -s https://raw.githubusercontent.com/kubestellar/kubestellar/release-$KUBESTELLAR_VERSION/test/e2e/common/setup-kubestellar-ocp.sh)
+   bash <(curl -s https://raw.githubusercontent.com/kubestellar/kubestellar/release-$KUBESTELLAR_VERSION/test/e2e/multi-cluster-deployment/use-kubestellar.sh) --env ocp
+fi
```
Comment (MikeSpreitzer) [inline test/e2e/multi-cluster-deployment/run-test.sh:60]: What should happen if `$env` is neither "kind" nor "ocp"? I think that a silent nothing is not a good answer. I think that an error message would be better.

File: test/e2e/common/cleanup.sh (hunk 1 above)
Comment (dumb0002) [reply test/e2e/common/cleanup.sh]: Fixed

File: test/e2e/common/cleanup.sh (hunk 2 above)
Comment (dumb0002) [reply test/e2e/common/cleanup.sh:22]: if the flag --env is missing, it would default to use its default parameter "kind". The e2e test for kind is already using the same script without providing any flags

File: test/e2e/multi-cluster-deployment/run-test.sh (hunk 3 above)
Comment (dumb0002) [reply test/e2e/multi-cluster-deployment/run-test.sh]: Fixed

File: test/e2e/multi-cluster-deployment/run-test.sh (hunk 5 above)
Comment (dumb0002) [reply test/e2e/multi-cluster-deployment/run-test.sh:63]: Fixed - added an error message

File: test/e2e/multi-cluster-deployment/run-test.sh (hunk 4 above)
Comment (dumb0002) [reply test/e2e/multi-cluster-deployment/run-test.sh:53]: This e2e test is focused to provide support to test KS releases on OCP - not a local copy.

Comment (ezrasilvera) [reply test/e2e/multi-cluster-deployment/run-test.sh:65]: @dumb0002 - correct. The testing that supports the on going development (i.e., tests that run on latest main for PR validation) are planed to be running only through the automatic github actions. 
Once we have the OCP tests hooked into the automatic CICD and decide we want to have OCP validation also for the ongoing development we can improve the tests to supprt that as well.

**Comment by ezrasilvera** [comment]
Thanks @dumb0002 . I put the LGTM.  I think you address all comments.  I will let you decide if/when to add the `approve`
/lgtm

File: test/e2e/multi-cluster-deployment/run-test.sh (hunk 4 above)
Comment (MikeSpreitzer) [reply test/e2e/multi-cluster-deployment/run-test.sh:65]: @dumb0002 : by "local copy" I mean the code copy, not where the containers run. The E2E tests have always been able to test a contributor's modified version of KubeStellar. This PR breaks that.
